### PR TITLE
feat(docs): add a /docs route using Lout

### LINKED ISSUE #108: Use Lout for route docs?
Not sure if this is useful or not (but may be handy if running in development/local dev mode). I don't know that I'm comfortable with this going into Prod as an endpoint (not sure why, it's easy to see the endpoints in source anyways), but yeah, we can probably put a handy switch and only include Lout if your `config.get('env')` is "local", "awsdev", or "test" in [`NODE_ENV`](https://github.com/mozilla/chronicle/blob/9767c8d0d595d98cf06e4ecb884951d543328b83/server/config.js#L13-L18) -- or just add a Boolean config flag for "showApiDocs".

Just for giggles, I did an `npm i lout -D` and then added the [Lout](https://github.com/hapijs/lout) plugin to our `server.register()` call (where `Lout` was `var Lout = require('lout');`'ed earlier):

```js
server.register([
  HapiAuthCookie,
  Bell,
  Lout
], function (err) {
```

Now when I goes to http://localhost:8080/docs I see the following sauce:
documentation_for_rest_routes
**Figure 1:** Route list

documentation_for
**Figure 2:** Route details

### PR DESCRIPTION
**Pull request by pdehaan**
Adds a <http://localhost:8080/docs> route which shows some auto-generated route documentation using [**Lout**](https://www.npmjs.com/package/lout).

It ain't pretty, but _meh_.

Closes #108

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -37,7 +38,8 @@ server.connection({
 
 server.register([
   HapiAuthCookie,
-  Bell
+  Bell,
+  Lout
```
Comment (pdehaan) [inline server/index.js]: Not sure about this bit right here.
I don't know if we should just blindly include Lout always, or if we want to wrap this in a `config.get('env')` and only include the /docs route if we're not in "production" environment.

**Comment by pdehaan** [comment]
r? @nchapman so he can look at how stupid this stupid is, and say "this is stupid, i don't want this" and close this.
Or merge it, whatever. Not like I'm paid per pull-request.

But yeah, I'm sure we could use some better descriptions, but figure we'll do baby steps and see if it's worth proceeding further before I try and copy/pasta [/docs/API.md](https://github.com/mozilla/chronicle/blob/master/docs/API.md) into various descriptions.

**Comment by pdehaan** [comment]
documentation_for_rest_routes

File: server/index.js (hunk 1 above)
Comment (6a68) [inline server/index.js]: I think it's fine to include it, and probably fine to load the lout route in production, too: our code's open source, we aren't really exposing anything.

**Comment by 6a68** [comment]
@pdehaan I'm game to try this, let's land it. I'm going to do a cleanup pass at the API this sprint (#222), seems like a good opportunity to give this a shot.

**Comment by 6a68** [comment]
Ah, looks like you'll need to rebase before landing. (If you want, you can start pushing branches to mozilla instead of your fork, then I can rebase-and-merge stuff like this for you.)

**Comment by pdehaan** [comment]
Rebased. Travis green. Self merge.

**Pull request merged**